[NUX]: Onboarding + settings pages

Sidsector9 opened the pull request:
### All Submissions:

<!-- Mark completed items with an [x] -->
* [ ] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:
<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #89, Closes #58.

This PR improves the merchant onboarding experiences and makes WooCommerce + Square integration and sync settings more intuitive.

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/e9ca8537-62af-4c4f-b27b-45e30e982157)

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/fb9e008c-7d1b-489b-95d7-d2c52d52bcbf)

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/86e09952-a4ed-41a1-9c8f-3513e695d273)

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/476661df-239e-4fc6-ac65-df0270e39915)

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/7151413b-aac8-4293-b231-6d2b1126d4ba)

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/6a3b8e82-16c6-434a-9f82-105b4c07d474)

![image](https://github.com/woocommerce/woocommerce-square/assets/25176325/da651fb7-05c5-4af3-91f2-0efed334c1d8)

### Steps to test the changes in this Pull Request:

The brand new onboarding wizard flow and settings pages should work as per [PT: NUX Improvements for Square](https://10upwoo.wordpress.com/2024/01/30/pt-nux-improvements-for-square/).

Notes:

- The "business location" step is displayed right after the successful connection and the "payment methods" page is displayed later.
- On the settings page, the business location field is moved after the connection field as it makes more sense.

### Changelog entry

> Add - New Merchant Onboarding experience with the new wizard flow & settings pages.

faisal-alvi committed:
add rest file for cash app

faisal-alvi committed:
init cash app rest file

faisal-alvi committed:
fill the reducer with cash app data

faisal-alvi committed:
a new selector for cash app

faisal-alvi committed:
an action for cash app

faisal-alvi committed:
a `saveCashAppSettings` util func

faisal-alvi committed:
hook `useCashAppData`

faisal-alvi committed:
include cash app as an onboarding step

faisal-alvi committed:
cash app index file step

faisal-alvi committed:
settings page for cash app gateway

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
set defaults when option is empty

faisal-alvi committed:
cash app gateway page fixes

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
added authorization dependent fields

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
fix cash app settings not saving issue

faisal-alvi committed:
Hide all elements after the react based settings container, hiding the native settings

faisal-alvi committed:
Merge branch '89-nux-onboarding' into wizard_page

faisal-alvi committed:
remove unnecessary icons

faisal-alvi committed:
remove the js files

faisal-alvi committed:
`ConnectSetup` onboarding step

faisal-alvi committed:
include the new step

faisal-alvi committed:
conditonally show the steps

faisal-alvi committed:
pass the step parameter

faisal-alvi committed:
add style for the connect to square onboarding page

faisal-alvi committed:
enqueue styles and scripts

faisal-alvi committed:
remove the enqueue funcs from main file

faisal-alvi committed:
move the basic funcs to settings file

faisal-alvi committed:
remove unnecessary HTML files

faisal-alvi committed:
redirect on activation and on successful connection

faisal-alvi committed:
fix the scripts/styles not enqueing issue

faisal-alvi committed:
fix wp-components style missing issue

faisal-alvi committed:
hide native fields from square settings page

faisal-alvi committed:
fix the typo

faisal-alvi committed:
link steps with modules

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
add square settings data store

Sidsector9 committed:
some helpful utils

Sidsector9 committed:
remove redundant method

Sidsector9 committed:
improve variable name

Sidsector9 committed:
add a save square settings button HOC

Sidsector9 committed:
push the business location page in the app

faisal-alvi committed:
remove usage of GET param

faisal-alvi committed:
show next page if connected already

faisal-alvi committed:
use the connection URL param

faisal-alvi committed:
add the connection URL param

faisal-alvi committed:
remove the logic for navigating b/w pages through GET param

faisal-alvi committed:
update the prop name

faisal-alvi committed:
conditional redirect on successful connection

faisal-alvi committed:
use the connection URL from store object

Sidsector9 committed:
add save button icon

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
add hook for payment gateway data

Sidsector9 committed:
add button to save payment gateway settings

Sidsector9 committed:
add payment gateway settings save buttons in onboarding

Sidsector9 committed:
save data on payment gateway toggles onboarding step

faisal-alvi committed:
add icons: sync and manage

faisal-alvi committed:
create payment complete onboarding step

faisal-alvi committed:
add the payment complete page in onboarding

faisal-alvi committed:
add CSS for onbaording wizard

faisal-alvi committed:
add CSS for settings section (react-based)

faisal-alvi committed:
manage CSS files across sections

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
create OnboardingHeader component

faisal-alvi committed:
style the onboarding component

faisal-alvi committed:
create more SVG icons

faisal-alvi committed:
include new SVG icons

faisal-alvi committed:
configure the Onboarding Header

faisal-alvi committed:
include OnboardingHeader in the list

faisal-alvi committed:
wrap input fields with a new `div`

faisal-alvi committed:
update step on SquareSettingsSaveButton click

faisal-alvi committed:
add required wrappers

faisal-alvi committed:
update step whenwithSaveSquareSettingsButton is hit

faisal-alvi committed:
Config Setting Module created

faisal-alvi committed:
remove config sync static, and use via component

faisal-alvi committed:
update next step on payment methods step

faisal-alvi committed:
create AdvancedSettings module and create a step for it

faisal-alvi committed:
use AdvancedSettings on settings page

faisal-alvi committed:
create SandboxSettings module

faisal-alvi committed:
create step for SandboxSettings

faisal-alvi committed:
remove the static, and use show SandboxSettings via componenet

faisal-alvi committed:
bind nextStep trigger with pay setting save btn

faisal-alvi committed:
remove repetative functions from util

faisal-alvi committed:
manage step and backStep in Onboarding flow

faisal-alvi committed:
remove unused hook

faisal-alvi committed:
remove unnecessary import

faisal-alvi committed:
Merge branch 'trunk' into wizard_page

faisal-alvi committed:
update the settings app

faisal-alvi committed:
Gift card app management accomplished

faisal-alvi committed:
Dynamic Loader Mechanism added

faisal-alvi committed:
Onboarding App management accomplished

faisal-alvi committed:
Cash app management accomplished

faisal-alvi committed:
Credit card management accomplished

faisal-alvi committed:
Import products popup created and configured

faisal-alvi committed:
Updated the whole new flow, where data traverse to avoid duplicate calls

faisal-alvi committed:
update the typography as per the design

faisal-alvi committed:
add fonts as per figma design

faisal-alvi committed:
show settings buttons only when enabled

faisal-alvi committed:
update the design as per figma

faisal-alvi committed:
add missing fields on CC settings

faisal-alvi committed:
save settings as per the page type

faisal-alvi committed:
save settings as per settings page type

faisal-alvi committed:
add enable logging field

faisal-alvi committed:
pass the data through param to save specific settings

faisal-alvi committed:
fix debug mode value

faisal-alvi committed:
add debug mode on CC page

faisal-alvi committed:
visit store button added

faisal-alvi committed:
pass required param for settings

faisal-alvi committed:
remove `useSettingsForm` as unused

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
remove snackbar in favour of in-button messages

Sidsector9 committed:
use hooks instead of props

Sidsector9 committed:
use hooks instead of props

Sidsector9 committed:
unify debug mode component

faisal-alvi committed:
adding `stepReducer`

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
use the step reduce data across components

Sidsector9 committed:
use hooks instead of props

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
useSteps as a separate hook

faisal-alvi committed:
Resolve conflicts,
Merge remote-tracking branch 'origin/wizard_page-use-steps' into wizard_page

faisal-alvi committed:
a few adjustments around useSteps usage

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
fix hook related bug when local storage for step is not set

Sidsector9 committed:
fix business location step

faisal-alvi committed:
remove prop drilling

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
Skip redirect existing users to the setup wizard on activation

Sidsector9 committed:
add afterSaveCallback

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
remove onboarding save button HOC

Sidsector9 committed:
fix saving button state

Sidsector9 committed:
fix setStep

faisal-alvi committed:
adjust the responsiveness around 13k-20k

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
add data-testid attributes

Sidsector9 committed:
add data-testid attributes

Sidsector9 committed:
wait for all settings to load

Sidsector9 committed:
add e2e tests for sync settings, cc and digital wallet

Sidsector9 committed:
add e2e tests for cash app and gift card

faisal-alvi committed:
final design (CSS) touches

faisal-alvi committed:
move the debug mode field

faisal-alvi committed:
add business location title

faisal-alvi committed:
minor feedback updates

faisal-alvi committed:
phpcs fixes across files

faisal-alvi committed:
a few more phpcs fixes

faisal-alvi committed:
fix the ignore statement

faisal-alvi committed:
phpcs ignore nonce

faisal-alvi committed:
Move & Migrate Declined Msg & Debug Mode Adv. Settings

faisal-alvi committed:
phpcs error fixes

Sidsector9 committed:
add utilities for toggle and minor refactor

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
Merge branch 'wizard_page' into onboarding-e2e

Sidsector9 committed:
add checks during sandbox connection

Sidsector9 committed:
Merge branch 'wizard_page' into onboarding-e2e

Sidsector9 committed:
update existing tests

Sidsector9 committed:
update existing tests

faisal-alvi committed:
show import products field for both

faisal-alvi committed:
fix styling

faisal-alvi committed:
minor phpcs fixes and since tag updates

faisal-alvi committed:
show notice about the new wizard and display menu page dynamically

faisal-alvi committed:
fix extra space

Sidsector9 committed:
disable import button when form is dirty

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
Merge branch 'wizard_page' into onboarding-e2e

Sidsector9 committed:
Merge branch 'onboarding-e2e' of github.com:woocommerce/woocommerce-square into onboarding-e2e

Sidsector9 committed:
update tests as per NUX changes

Sidsector9 committed:
fix function sequence

Sidsector9 committed:
fix gift cards

Sidsector9 committed:
Merge branch 'wizard_page' into onboarding-e2e

Sidsector9 committed:
update tests related to card logos

Sidsector9 committed:
add tests for steps

dkotter commented on assets/new-user-experience/components/onboarding-header/index.js:
```diff
@@ -0,0 +1,55 @@
+export const OnboardingHeader = () => {
+    const {
+		setStep,
+        getBackStep,        
```
Looks like some weird spacing here

dkotter commented on assets/new-user-experience/icons/back.js:
```diff
@@ -0,0 +1,7 @@
+export const Back = () => {
+	return (
```
Looks like an extra tab here or a mix of tabs and spaces in this file

dkotter commented on assets/new-user-experience/icons/close.js:
```diff
@@ -0,0 +1,7 @@
+export const Close = () => {
+	return (
```
Looks like an extra tab here or a mix of tabs and spaces in this file

dkotter commented on assets/new-user-experience/icons/manage.js:
```diff
@@ -0,0 +1,8 @@
+export const Manage = () => {
+	return (
```
Looks like an extra tab here or a mix of tabs and spaces in this file

dkotter commented on assets/new-user-experience/icons/square.js:
```diff
@@ -0,0 +1,14 @@
+export const Square = () => {
+	return (
```
Looks like an extra tab here or a mix of tabs and spaces in this file

dkotter commented on assets/new-user-experience/icons/sync.js:
```diff
@@ -0,0 +1,8 @@
+export const Sync = () => {
+	return (
```
Looks like an extra tab here or a mix of tabs and spaces in this file

dkotter commented on assets/new-user-experience/modules/advanced-settings/index.js:
```diff
@@ -0,0 +1,90 @@
+		debug_logging_enabled = 'no',
+	} = settings;
+
+	if ( ! ( squareSettingsLoaded ) ) {
```
Do we need the extra parentheses here?
```suggestion
	if ( ! squareSettingsLoaded ) {
```

dkotter commented on assets/new-user-experience/modules/advanced-settings/index.js:
```diff
@@ -0,0 +1,90 @@
+							checked={ 'yes' === enable_customer_decline_messages }
+							onChange={ ( enabled ) => setSquareSettingData( { enable_customer_decline_messages: enabled ? 'yes' : 'no' } ) }
+							label={
+								__( 'Check to enable detailed decline messages to the customer during checkout when possible, rather than a generic decline message.', 'woocommerce-square' )
```
Just my opinion but seems pretty obvious that they need to check this option to turn it on so not sure we need that in a label/description. I'd do something like:
```suggestion
								__( 'Show detailed decline messages to the customer during checkout rather than a generic decline message.', 'woocommerce-square' )
```

dkotter commented on assets/new-user-experience/modules/advanced-settings/index.js:
```diff
@@ -0,0 +1,90 @@
+					<InputWrapper
+						label={ __( 'Enable Logging', 'woocommerce-square' ) }
+						variant="boxed"
+						description={ __( 'After enabling you’ll see a new Sandbox settings section with two fields; Sandbox Application ID & Sandbox Access Token.', 'woocommerce-square' )}
```
```suggestion
						description={ __( 'After enabling you’ll see a new Sandbox settings section with two fields: Sandbox Application ID & Sandbox Access Token.', 'woocommerce-square' )}
```

dkotter commented on assets/new-user-experience/modules/configure-sync/index.js:
```diff
@@ -0,0 +1,304 @@
+		is_connected = false,
+	} = settings;
+
+	const sync_interval_options = [
```
I've not gotten further yet so maybe I'll find the answer to this later but if I recall correctly, we have a PHP filter right now allowing these options to be changed. Is that something that can still be done in this new experience?

faisal-alvi replied:
We do not have a filter for options but I can see a filter in PHP `wc_square_sync_interval` to filter the selected interval. We are not using that in this new react-based settings page. To do so we may need to create an endpoint to get the value from the PHP filter into the JS file.

dkotter commented on assets/new-user-experience/modules/sandbox-settings/index.js:
```diff
@@ -0,0 +1,87 @@
+				<div className='woo-square-wizard__fields'>
+					<InputWrapper
+						label={ __( 'Enable Sandbox Mode', 'woocommerce-square' ) }
+						description={ __( "After enabling you'll see a new Sandbox settings section with two fields; Sandbox Application ID & Sandbox Access Token.", 'woocommerce-square' ) }
```
```suggestion
						description={ __( "After enabling you'll see a new Sandbox settings section with two fields: Sandbox Application ID & Sandbox Access Token.", 'woocommerce-square' ) }
```

dkotter commented on assets/new-user-experience/modules/sandbox-settings/index.js:
```diff
@@ -0,0 +1,87 @@
+						<>
+							<InputWrapper
+								label={ __( 'Sandbox Application ID', 'woocommerce-square' ) }
+								description={ __( 'Application ID for the Sandbox Application, see the details in the My Applications section.', 'woocommerce-square' ) }
```
Could we link to the correct spot, the `My Applications` page?

dkotter commented on assets/new-user-experience/modules/sandbox-settings/index.js:
```diff
@@ -0,0 +1,87 @@
+
+							<InputWrapper
+								label={ __( 'Sandbox Access Token', 'woocommerce-square' ) }
+								description={ __( 'Access Token for the Sandbox Test Account, see the details in the Sandbox Test Account section. Make sure you use the correct Sandbox Access Token for your application. For a given Sandbox Test Account, each Authorized Application is assigned a different Access Token.', 'woocommerce-square' ) }
```
Same here, would be great to link to the right spot if we can

dkotter commented on assets/new-user-experience/onboarding/data/reducers.js:
```diff
@@ -0,0 +1,201 @@
+export const CREDIT_CARD_DEFAULT_STATE = {
+	enabled: 'yes',
+	title: 'Credit Card',
```
There's a decent amount of text in this file that isn't being translated. I've not parsed everything yet so maybe that doesn't matter but flagging that as something to consider if needed.

dkotter commented on assets/new-user-experience/onboarding/hooks/index.js:
```diff
@@ -0,0 +1,210 @@
+	const setBackStep = ( data ) => dispatch( store ).setBackStep( data );
+
+	const stepData = {
+		// step: localStorage.getItem('step') || getStep(),
+		// backStep: localStorage.getItem('backStep') || getBackStep(),
```
Assuming we can remove this

dkotter commented on assets/new-user-experience/onboarding/onboarding-app.js:
```diff
@@ -0,0 +1,215 @@
+	} = useSquareSettingsData;
+
+	// Set info in local storage.
+	useEffect(() => {
```
Seeing a mix of styling in this file, some places don't have spaces (like here) but others do. We should standardize that to match what we're doing elsewhere (looks like adding spaces is the desired outcome)

dkotter commented on assets/new-user-experience/onboarding/steps/business-location/index.js:
```diff
@@ -0,0 +1,108 @@
+				<SectionTitle title={ __( 'Your Square account is missing a Business Location', 'woocommerce-square' ) } />
+				<SectionDescription>
+					<p>
+						{ __( 'Please go here or use the button below to create a Business Location and then return to WooCommerce to complete setup.', 'woocommerce-square' ) }
```
Are the words `go here` supposed to be linked somewhere?

faisal-alvi replied:
Yes, it is not specified where in the Figma though. I added https://squareup.com/dashboard/locations

dkotter commented on assets/new-user-experience/onboarding/steps/business-location/index.js:
```diff
@@ -0,0 +1,108 @@
+	return (
+		<div>
+			<Section>
+				{(locationCount === 0 && noLocation) ||
```
Same question here in regards to spacing. I see a few places here and later in the file that don't have spaces while the rest of the code does

faisal-alvi replied:
I haved added spaces so it is now:

`{ ( locationCount === 0 && noLocation ) ||`

But the following seems okay (without spaces), please let me know if this should also be changed.

`{intro}` and `options={[`

dkotter commented on assets/new-user-experience/onboarding/steps/digital-wallets/index.js:
```diff
@@ -0,0 +1,156 @@
+			<Section>
+				<SectionTitle title={ __( 'Manage Digital Wallet Settings', 'woocommerce-square' ) } />
+				<SectionDescription>
+					{ __( 'Accept payments with Apple Pay and Google Pay on your store, available in select countries. Enabling digital wallets adds payment buttons to Product, Cart, and Checkout pages.', 'woocommerce-square' ) }
```
```suggestion
					{ __( 'Accept payments with Apple Pay and Google Pay on your store, available in select countries. Enabling digital wallets adds payment buttons to Product, Cart and Checkout pages.', 'woocommerce-square' ) }
```

dkotter commented on assets/new-user-experience/onboarding/steps/payment-complete/index.js:
```diff
@@ -0,0 +1,184 @@
+				<div className="woo-square-onbarding__payment-settings--left">
+					<div className="woo-square-onbarding__payment-settings__intro">
+						<div className="woo-square-onbarding__payment-settings__intro--title">
+							{ __( "Congratulations,", 'woocommerce-square' ) }
+							<br />
+							{ __( "Your Payment Setup is Complete!", 'woocommerce-square' ) }
+						</div>
+						<SectionDescription>
+							{ __( "Your online store is now equipped to accept payments, making you ready to welcome customers and start generating sales.", 'woocommerce-square' ) }
```
Seems like all these text strings can be wrapped in single quotes instead of double

dkotter commented on assets/new-user-experience/styles/fonts/sf-pro/SFPRODISPLAYBLACKITALIC.OTF:
Curious if all of these fonts are actually used? Would prefer not to bundle any that aren't but maybe all of them are needed

dkotter commented on assets/new-user-experience/styles/index.scss:
```diff
@@ -0,0 +1,205 @@
+// import fonts SFPRODISPLAYREGULAR.OTF from /Users/faisalalvi/LocalSites/wpne/app/public/wp-content/plugins/woocommerce-square/assets/new-user-experience/fonts/sf-pro/SFPRODISPLAYREGULAR.OTF
```
Assuming we can remove this

dkotter commented on assets/onboarding/components/index.js:
Do we need this file? Says it's empty

dkotter commented on assets/onboarding/elements/index.js:
Do we need this file? Says it's empty

dkotter commented on assets/onboarding/hooks/index.js:
Do we need this file? Says it's empty

dkotter commented on assets/onboarding/steps/index.js:
Do we need this file? Says it's empty

dkotter commented on includes/Admin/Rest/WC_REST_Square_Cash_App_Settings_Controller.php:
```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Class WC_REST_Square_Settings_Controller
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Cash_App_Settings_Controller.php:
```diff
@@ -0,0 +1,157 @@
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for saving Square's test/live account keys.
+ *
+ * This includes Live Publishable Key, Live Secret Key, Webhook Secret.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Gift_Cards_Settings_Controller.php:
```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Class WC_REST_Square_Settings_Controller
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Gift_Cards_Settings_Controller.php:
```diff
@@ -0,0 +1,127 @@
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for saving Square's test/live account keys.
+ *
+ * This includes Live Publishable Key, Live Secret Key, Webhook Secret.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Credit_Card_Payment_Settings_Controller.php:
```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Class WC_REST_Square_Settings_Controller
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Credit_Card_Payment_Settings_Controller.php:
```diff
@@ -0,0 +1,189 @@
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for saving Square's test/live account keys.
+ *
+ * This includes Live Publishable Key, Live Secret Key, Webhook Secret.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Credit_Card_Payment_Settings_Controller.php:
```diff
@@ -0,0 +1,189 @@
+	private $allowed_params;
+
+	/**
+	 * WC_REST_Square_Settings_Controller constructor.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Credit_Card_Payment_Settings_Controller.php:
```diff
@@ -0,0 +1,189 @@
+	}
+
+	/**
+	 * Retrieve flag status.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Cash_App_Settings_Controller.php:
```diff
@@ -0,0 +1,157 @@
+	private $allowed_params;
+
+	/**
+	 * WC_REST_Square_Settings_Controller constructor.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Cash_App_Settings_Controller.php:
```diff
@@ -0,0 +1,157 @@
+	}
+
+	/**
+	 * Retrieve flag status.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Gift_Cards_Settings_Controller.php:
```diff
@@ -0,0 +1,127 @@
+	private $allowed_params;
+
+	/**
+	 * WC_REST_Square_Settings_Controller constructor.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Gift_Cards_Settings_Controller.php:
```diff
@@ -0,0 +1,127 @@
+	}
+
+	/**
+	 * Retrieve flag status.
```
Needs updating

dkotter commented on includes/Admin/Rest/WC_REST_Square_Settings_Controller.php:
```diff
@@ -0,0 +1,218 @@
+	}
+
+	/**
+	 * Retrieve flag status.
```
Needs updating

dkotter commented on includes/Framework/PaymentGateway/Payment_Gateway.php:
```diff
@@ -341,6 +338,12 @@ public function __construct( $id, $plugin, $args ) {
 
+		$square_settings = get_option( 'wc_square_settings', array() );
+
+		$this->debug_mode = isset( $square_settings['debug_mode'] ) ? $square_settings['debug_mode'] : 'off';
+
+		$this->enable_customer_decline_messages = isset( $square_settings['enable_customer_decline_messages'] ) ? $square_settings['enable_customer_decline_messages'] : 'no';
```
I think both of these can be simplified using the `??` operator

dkotter commented on includes/Framework/PaymentGateway/Payment_Gateway_Plugin.php:
```diff
@@ -648,7 +647,11 @@ protected function add_debug_setting_notices() {
-			if ( $gateway->is_enabled() && $gateway->is_production_environment() && ! $gateway->debug_off() ) {
+			// Get the debug mode.
+			$square_settings = get_option( 'wc_square_settings', array() );
+			$debug_mode      = isset( $square_settings['debug_mode'] ) ? $square_settings['debug_mode'] : 'off';
```
```suggestion
			$debug_mode      = $square_settings['debug_mode'] ?? 'off';
```

dkotter commented on includes/Handlers/Products.php:
```diff
@@ -75,6 +78,10 @@ public function __construct( Square\Plugin $plugin ) {
 
+		// Get gift card features status.
+		$gift_card_settings      = get_option( 'woocommerce_gift_cards_pay_settings', array() );
+		$this->gift_card_enabled = isset( $gift_card_settings['enabled'] ) ? $gift_card_settings['enabled'] : 'no';
```
```suggestion
		$this->gift_card_enabled =$gift_card_settings['enabled'] ?? 'no';
```

dkotter commented on includes/Lifecycle.php:
```diff
@@ -293,6 +294,45 @@ protected function upgrade_to_3_0_2() {
 	}
 
+	/**
+	 * Upgrades to version 2.3.0.
```
```suggestion
	 * Upgrades to version 4.7.0.
```

dkotter commented on includes/Lifecycle.php:
```diff
@@ -293,6 +294,45 @@ protected function upgrade_to_3_0_2() {
+		);
+
+		// Mark upgrade complete.
+		update_option( 'wc_square_updated_to_4_7_0', true );
+
+		// Skip redirect existing users to the setup wizard on upgrade.
+		add_option( 'wc_square_show_wizard_on_activation', true );
```
Should we ensure these options aren't autoloaded?

dkotter commented on includes/Settings.php:
```diff
@@ -126,6 +139,124 @@ function( $fields ) {
+	 * @since 4.7.0
+	 */
+	public function show_visit_wizard_notice() {
+		if ( ! get_option( 'wc_square_onboarding_wizard_visited' ) ) {
```
Instead of having this larger block of code nested, we could invert this condition:
```suggestion
		if ( get_option( 'wc_square_onboarding_wizard_visited' ) ) {
			return;
		}
```

dkotter commented on includes/Settings.php:
```diff
@@ -829,8 +983,14 @@ public function get_location_id() {
 	 */
 	public function get_locations() {
 
+		$is_ajax = wp_doing_ajax();
```
Now that this method is called by ajax, do we need any nonce verification here?

dkotter commented on woocommerce-square.php:
```diff
@@ -29,7 +29,7 @@
 
 if ( ! defined( 'WC_SQUARE_PLUGIN_VERSION' ) ) {
-	define( 'WC_SQUARE_PLUGIN_VERSION', '4.6.0' ); // WRCS: DEFINED_VERSION.
+	define( 'WC_SQUARE_PLUGIN_VERSION', '4.7.0' ); // WRCS: DEFINED_VERSION.
```
```suggestion
	define( 'WC_SQUARE_PLUGIN_VERSION', '4.6.1' ); // WRCS: DEFINED_VERSION.
```

dkotter reviewed (changes requested):
Let me know if there's any questions on the items I've flagged.

One other question I have is that this PR is set to merge into the `nux-react` branch. Curious on what this branch is and if it's still needed, or if this should be changed to `trunk`? cc / @Sidsector9 @faisal-alvi

faisal-alvi committed:
suggestions implemented

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
Merge branch 'trunk' into wizard_page

faisal-alvi committed:
more suggestions implemented; nonce, translations, etc.

faisal-alvi committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

faisal-alvi committed:
move the nonce

faisal-alvi commented:
@dkotter thanks for the feedback. I've implemented the requested changes/suggestions and added some inline comments too. Please let me know if any changes are required. Note: All the spaces in the JS files are converted into tabs.

> this PR is set to merge into the nux-react branch. Curious on what this branch is and if it's still needed, or if this should be changed to trunk?

Yes, I changed the base to `trunk`.

> Do we need this file? Says it's empty

I can see the files are not empty. I'm not sure how this is empty at your end.

dkotter commented:
> I can see the files are not empty. I'm not sure how this is empty at your end.

Not seeing this anymore, wondering if this was due to the base branch being `nux-react` instead of `trunk` but in either case, good now

faisal-alvi committed:
Merge branch 'wizard_page' into onboarding-e2e

faisal-alvi committed:
remove unnecessary code

faisal-alvi committed:
fix `connection_url_wizard` not found issue

faisal-alvi committed:
Merge branch 'wizard_page' into onboarding-e2e

qasumitbagthariya commented:
Hello @faisal-alvi 

Thanks for the PR.

In my initial testing, I found some CSS issues on the Chrome browser. Can you please check on your end?

**Screen resolution 1920*1080** 
![image](https://github.com/woocommerce/woocommerce-square/assets/67687255/5ec4c410-c070-4ac7-a8fc-5ce67600849f)

**Screen resolution 1440*900**
<img width="1435" alt="image" src="https://github.com/woocommerce/woocommerce-square/assets/67687255/b9e3e603-229d-448f-89be-1471ebd2ccf0">

**Screen resolution 1920*1080** 
![image](https://github.com/woocommerce/woocommerce-square/assets/67687255/2af8f70f-48f7-429e-9142-4e3235df9872)

I also noticed there are two scroll bars on the screen.
![image](https://github.com/woocommerce/woocommerce-square/assets/67687255/df08d714-b647-4ad1-99d0-58e8780af34d)

**Note:**
⚠️  My testing is still in progress.

faisal-alvi committed:
QA feedback design fixes

faisal-alvi committed:
font size issue, converted to woff2

faisal-alvi committed:
show business page button label conditionally

faisal-alvi committed:
update the copies; centered logo

qasumitbagthariya commented:
Hello @faisal-alvi 

I've created a bug sheet where I'll add all the bugs, to avoid back and forth so that we can track all the bugs and their status. 

**Bug sheet:**
https://docs.google.com/spreadsheets/d/1OpDrRRWhpUmxokCGPYQasPjUyQ9HJGkxqVOc3oUqL7w/edit#gid=0

Note:
⚠️ My testing is still in progress.

Thanks.

faisal-alvi committed:
fix PHP Deprecated error

faisal-alvi committed:
fix alignment + parse HTML

qasumitbagthariya commented:
@faisal-alvi @Sidsector9  ⚠️

When setting up a wizard (fresh setup) in the "Select your business location" section, it appears to be selecting a location, but it actually selects a Square application. After completing all the onboarding setups, when attempting to set up again(from the plugin page), it shows the business location. This inconsistency poses an issue. It's possible that we need to add one more screen, like selecting an application from Square, to address this.

Let me know if you need any help from my side to reproduce this flow.


https://github.com/woocommerce/woocommerce-square/assets/67687255/c49661ef-da5a-4f82-ad36-92a7b6d2c6c2


https://github.com/woocommerce/woocommerce-square/assets/67687255/ddacfc7a-09e8-4d07-96ca-e1f28dee40cb

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
disable sandbox by default

Sidsector9 committed:
fix build process

Sidsector9 commented:
@qasumitbagthariya as discussed on Slack, this is good to go 👍

qasumitbagthariya commented:
### QA Update ✅ 

-------------
 I have verified [NUX] Onboarding + Setting page in the `wizard_page` branch and functioning as intended, all the [issues](https://docs.google.com/spreadsheets/d/1OpDrRRWhpUmxokCGPYQasPjUyQ9HJGkxqVOc3oUqL7w/edit#gid=0) are resolved.

I tested the following on this branch:
- Onboarding + Setting page ✅ 
- Plugin > Square > Setup wizard ✅ 
- Compare all screens with figma ✅ 
- Gift card should not display if cc is off ✅  
- migration (Detailed Decline Messages and debug mode, Giftcard) ✅ 
- Cashapp ✅ 
- Credit payment ✅ 
- Import products ✅ 
- sync products ✅ 


https://github.com/woocommerce/woocommerce-square/assets/67687255/5f8cb06d-b00e-443c-ba25-b5fa0dc0a22d


https://github.com/woocommerce/woocommerce-square/assets/67687255/1cda83c8-1447-4c85-8ec2-e7424172fd75





**Testing Environment** 

<Details>

- WordPress: 6.5.3
- Theme: Storefront 4.5.5
- Theme: Twenty Twenty-Four 1.1
- WooCommerce - 8.9.0
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 

</Details>

Steps to Test- As mentioned in the PR description.
Test Results - It is working as expected.
Functional Demo / Screencast - 
Special Notes -  Ready for code review (Woo)
Testing Document status:
Cases related to this Issue/PR are added to the Critical Flow Wiki pages: 

- [x] Yes
- [ ] Not Required/Applicable for this PR

Sidsector9 committed:
Merge branch 'onboarding-e2e' of github.com:woocommerce/woocommerce-square into onboarding-e2e

Sidsector9 committed:
Merge branch 'wizard_page' into onboarding-e2e

Sidsector9 committed:
Merge pull request #109 from woocommerce/onboarding-e2e

[NUX]: E2E tests

vikrampm1 commented:
@mattallan noting that this PR is awaiting your review.

Sidsector9 committed:
Merge branch 'trunk' into wizard_page

mattallan reviewed (commented):
Kudos @Sidsector9 for working on this change! Nice changes/improvements coming in this one which is cool to see!

I've tested this and noticed a couple of things:

### Square Onboarding on local dev site

When I pulled the latest branch on my local dev site, I clicked on Square onboarding, logged into my Square account and then after being redirected to my local store, I was basically stuck on this page:

![image](https://github.com/woocommerce/woocommerce-square/assets/2275145/fca5b506-fc04-41f8-94ef-f4665be683c1)

A few issues I notice:
1. The HTML link is being printed
2. The button was taking me to `squareup.com/dashboard` instead of `squareup.com/dashboard/locations`
3. My Square account already has locations available

> My Square account already has locations available

Regarding this point ^ Does this problem only exist when testing on a local server? When I created a live site, going through the onboarding wizard worked perfectly fine 🤔 

### Debug notice linking to wrong page

This notice sends the merchant to the **WooCommerce > Settings > Payments > Square** page but there's no debug mode setting there anymore:

![image](https://github.com/woocommerce/woocommerce-square/assets/2275145/db0c1b9d-bd1b-4dec-8557-3989c9dfde7c)

This setting was moved to **WooCommerce > Settings > Square**.

Something I noticed is there are now two settings related to debugging (Debug mode and enable logging): 
![image](https://github.com/woocommerce/woocommerce-square/assets/2275145/b74a7ed9-556d-4fb3-a95c-811ab885a7e0)

Previously these two settings had different functionalities:
1. Debug mode was specific to showing payment gateway related errors
2. Enable logging was specific to logging import and syncing errors

Is this distinction still the case? If so, I wonder if we should clarify this now that they're both on the same settings page?

### Existing store after updating

On my existing store that is already connected to Square and has orders/subscriptions purchased with Square, when I updated to this new version, I'm noticed the Square Onboarding menu item is loading:

![image](https://github.com/woocommerce/woocommerce-square/assets/2275145/2f6aace5-6852-45a1-8552-687381d7dee9)

Clicking on it sent me directly to a "Successfully connected" page without me having to do anything but I'm wondering if this should be automatic? Is it intended that all existing merchants see this onboarding wizard? Could we set the flag on upgrade.

### Viewing Square settings on local dev site

I'm not sure what's going on here but whenever I view a Square settings page on my local dev site I'm getting a gap at the top of the page. Here's a quick recording:

https://github.com/woocommerce/woocommerce-square/assets/2275145/1c9d4797-3438-4bed-b91f-6827edcb1785

I'm not seeing this on my live site and I don't have any PHP errors in logs or JS errors so I'm really not sure what would be causing this but I just wanted to raise it in case anyone else has seen it before.

Sidsector9 committed:
Merge branch 'wizard_page' of github.com:woocommerce/woocommerce-square into wizard_page

Sidsector9 committed:
fix business locations formatting

Sidsector9 commented:
Thanks for reviewing @mattallan 

> Regarding this point ^ Does this problem only exist when testing on a local server? When I created a live site, going through the onboarding wizard worked perfectly fine 🤔

It should work locally, with and w/o https. Can you share the details of your stack you use for local development? It'll help us debug it further.

mattallan commented:
Hey @Sidsector9, I've just been trying to reproduce it again locally but not having any luck 😢
It only happened straight after I updated Square and it's been a while since I've used Square locally so I'm wondering if I had some old invalid/expired Square keys still saved, and so after going through the wizard and being redirected back to the site it couldn't fetch any locations?

I've just been trying to reproduce similar issues to what I saw using that theory but still no luck, so I'm not sure. Sorry about the false report!

> Can you share the details of your stack you use for local development?

Yeah, it's a standard [Laravel Valet](https://laravel.com/docs/11.x/valet) local site, PHP 8.2, WC 8.8.3. Here's the SSR if that's easier:

<details>
`
### WordPress Environment ###

WordPress address (URL): http://woo.localhost
Site address (URL): http://woo.localhost
WC Version: 8.8.3
REST API Version: ✔ 8.8.3
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.5.3
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: –
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.25.4
PHP Version: 8.2.16
PHP Post Max Size: 512 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 8.6.0
(SecureTransport) OpenSSL/3.2.1

SUHOSIN Installed: –
MySQL Version: 5.7.44
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.8.3
WC Database Prefix: wp_
Total Database Size: 48.77MB
Database Data Size: 32.49MB
Database Index Size: 16.28MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.33MB + Index: 0.16MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 3.52MB + Index: 3.92MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 1.02MB + Index: 2.41MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 3.52MB + Index: 4.20MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_events: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 2.52MB + Index: 1.56MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_newsletter: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_newsletter_emails: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_newsletter_sent: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_newsletter_stats: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_newsletter_user_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 5.02MB + Index: 0.16MB + Engine InnoDB
wp_postmeta: Data: 2.05MB + Index: 1.16MB + Engine InnoDB
wp_posts: Data: 0.14MB + Index: 0.06MB + Engine InnoDB
wp_rank_math_internal_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_rank_math_internal_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.22MB + Index: 0.19MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wcs_payment_retries: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_bundle_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points_log: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_subscription_downloads: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpforms_payments: Data: 0.02MB + Index: 0.14MB + Engine InnoDB
wp_wpforms_payment_meta: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wpforms_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpml_mails: Data: 12.52MB + Index: 0.00MB + Engine InnoDB
wp_wp_phpmyadmin_extension__errors_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

acf-field-group: 1
attachment: 42
aw_workflow: 4
customize_changeset: 6
dummy_subscription: 5
global_product_addon: 1
page: 18
post: 11
product: 39
product_variation: 34
revision: 38
shop_coupon: 3
shop_order: 20
shop_subscription: 11
tribe_events: 1
tribe_venue: 1
wp_global_styles: 2
wp_navigation: 1
wp_template: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

WooCommerce Square: by WooCommerce – 4.6.1
WooCommerce: by Automattic – 8.8.3

### Inactive Plugins (58) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.1
AutomateWoo: by WooCommerce – 5.5.5 (update to version 6.0.23 is available)
Braintree for WooCommerce Payment Gateway: by WooCommerce – 3.0.6 (update to version 3.1.5 is available)
Custom Functions Plugin: by Shamim – 1.1
Custom Start Date For WooCommerce Subscriptions: by Launch & Sell – 1.3.12
eCommerce Configuration By WP Engine: by WP Engine – 0.2.11
Gutenberg: by Gutenberg Team – 18.3.0
Jetpack: by Automattic – 13.4.2
Jetpack Boost: by Automattic - Jetpack Site Speed team – 3.3.0
Query Monitor: by John Blackbourn – 3.15.0
Relative URL: by Tunghsiao Liu – 0.1.8
Sensei Pro (WooCommerce Paid Courses): by Automattic – 4.0.2.2.6.4 (update to version 4.24.0.1.24.0 is available)
Smush: by WPMU DEV – 3.14.1
The Events Calendar: by The Events Calendar – 5.16.2.1
THSA Apps | One-click Switcher: by The Health Sciences Academy – 2.0.7
Variations as Single Product for WooCommerce: by StorePlugin – 1.0.0
WCPay Subscription Generator: by James Allan –
Woo All Products For Subscriptions: by Woo – 5.0.3 (update to version 5.0.4 is available)
WooCommerce Admin: by WooCommerce – 3.3.2
WooCommerce Affirm Gateway: by WooCommerce – 2.2.0 (update to version 2.3.0 is available)
WooCommerce Amazon Pay: by WooCommerce – 2.5.1
WooCommerce Blocks: by Automattic – 7.5.0
WooCommerce Bookings: by WooCommerce – 1.15.48 (update to version 2.1.2 is available)
WooCommerce Currency Converter: by WooCommerce – 1.6.25 (update to version 2.2.3 is available)
WooCommerce Deposits: by WooCommerce – 2.2.5 (update to version 2.2.8 is available)
WooCommerce Dynamic Pricing: by Element Stark – 3.1.26 (update to version 3.2.7 is available)
WooCommerce EU VAT Number: by WooCommerce – 2.9.0 (update to version 2.9.3 is available)
WooCommerce Eway Payment Gateway: by WooCommerce – 3.4.1 (update to version 3.7.2 is available)
WooCommerce GoCardless Gateway: by WooCommerce – 2.5.9 (update to version 2.6.4 is available)
WooCommerce Memberships: by SkyVerge – 1.22.9 (update to version 1.26.5 is available)
WooCommerce Payfast Gateway: by WooCommerce – 1.6.5
WooCommerce Payments Dev Tools: by Automattic –
WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.1.3
WooCommerce PayPal Payments: by WooCommerce – 2.7.1
WooCommerce Points and Rewards: by WooCommerce – 1.6.43 (update to version 1.8.2 is available)
WooCommerce Pre-Orders: by WooCommerce – 2.1.0
WooCommerce Product Add-ons: by WooCommerce – 6.4.7 (update to version 6.8.3 is available)
WooCommerce Purchase Order Payment Gateway: by WooCommerce – 1.4.3 (update to version 1.4.7 is available)
WooCommerce Shipment Tracking: by WooCommerce – 1.6.31 (update to version 2.4.8 is available)
WooCommerce SnapScan Gateway: by WooCommerce – 1.2.7
WooCommerce Square Helper: by WooCommerce – 1.0.5
WooCommerce Stripe Gateway: by WooCommerce – 8.2.0
WooCommerce Subscription Downloads: by WooCommerce – 1.4.1
WooCommerce Subscriptions: by WooCommerce – 6.3.1
WooCommerce Subscriptions - Custom Price String: by WooCommerce – 1.0.6
WooCommerce Subscriptions Core: by Automattic – 7.1.1
WooCommerce Subscriptions CSV Importer and Exporter: by Prospress Inc – 2.1.0
WooCommerce Subscriptions Gifting: by WooCommerce – 2.7.0
WooPayments: by Automattic – 7.6.0
Woo Product Bundles: by Woo – 7.0.1 (update to version 7.1.1 is available)
Woo Subscriptions Gifting - Recipient Address Update Fix: by Matt Allan –
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.5
WP Console: by Edi Amin – 2.4.1
WP Crontrol: by John Blackbourn – 1.16.2
WPForms Lite: by WPForms – 1.8.3.1
WP Mail Logging: by WP Mail Logging Team – 1.12.0
WP Mail SMTP: by WP Mail SMTP – 3.8.0
WP phpMyAdmin: by Puvox.software – 5.2.1.02

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
new-variable (new-variable)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 78 MB

### WC Pages ###

Shop base: #1789 - /shop/
Cart: #7 - /cart/
Checkout: #941 - /checkout-block/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.5.4 (update to version 4.6.0 is available)
Author URL: https://woo.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
product-pre-publish-modal
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
product-custom-fields
settings
async-product-editor-category-field
launch-your-store

Daily Cron: ✔ Next scheduled: 2023-08-23 03:27:13 +00:00
Options: ✔
Notes: 84
Onboarding: skipped

### Action Scheduler ###

Complete: 355
Oldest: 2024-04-09 00:10:13 +0000
Newest: 2024-06-02 00:58:14 +0000

Pending: 112
Oldest: 2024-05-08 01:27:53 +0000
Newest: 2026-05-20 03:02:57 +0000

Failed: 60
Oldest: 2024-02-09 05:36:16 +0000
Newest: 2024-06-02 00:58:11 +0000

Canceled: 910
Oldest: -0001-11-30 00:00:00 +0000
Newest: -0001-11-30 00:00:00 +0000


### Status report information ###

Generated at: 2024-05-21 08:37:00 +00:00
`
</details>

faisal-alvi commented:
Hey @mattallan thanks for the report.

> Is this distinction still the case? If so, I wonder if we should clarify this now that they're both on the same settings page?

Yes, it is still the case. We can add a description to the settings field saying that enabling this setting will also enable debugging for other gateways/payment methods. How does that sound? However, I believe that seeing these fields on the generic settings page itself makes it obvious that they will be used globally for all payment gateways/methods.

> Clicking on it sent me directly to a "Successfully connected" page without me having to do anything but I'm wondering if this should be automatic?

After WP v6.5, on plugin activation, the auto-redirect is not supported anymore.

> Is it intended that all existing merchants see this onboarding wizard? Could we set the flag on upgrade.

Yes, we display the notice about the new onboarding wizard on upgrade to the existing users. It disappears when a user visits the onboarding wizard.

https://github.com/woocommerce/woocommerce-square/blob/367f4c3765557e14a1cb1cb610a2d86c9e6a2e93/includes/Settings.php#L216

> I'm getting a gap at the top of the page

I checked and it does not appear to me. Can you please confirm if there is any PHP warning behind that white strip?

mattallan commented:
Thanks @faisal-alvi for the response!

> We can add a description to the settings field saying that enabling this setting will also enable debugging for other gateways/payment methods. How does that sound? However, I believe that seeing these fields on the generic settings page itself makes it obvious that they will be used globally for all payment gateways/methods.

Hmm, this comment makes me think I might be misunderstanding how these two settings work now 😅 

![image](https://github.com/woocommerce/woocommerce-square/assets/2275145/25949937-ad67-4a56-b728-d6083ed692d4)

From my understanding, these two settings are independent of each other and have different functionalities:

1. Debug Mode: Toggles logging Square payment gateway errors on/off
2. Enable Logging: Toggles logging Square connection, import & sync errors on/off

Given this, merchants can have "Enable Logging" turned off and have Debug mode set to "Save to log" and errors from the checkout will still be logged. For me, this isn't clear from looking at these two settings which is why I was thinking that adding a description will help avoid merchant confusion

My understanding of these two settings might be incorrect now though so I'm happy for you to confirm and decide on whether a description is needed.

> Yes, we display the notice about the new onboarding wizard on upgrade to the existing users.

I wasn't referring to the notice, I was referring to the new **Square Onboarding** menu item found under the WooCommerce side menu 🙂 I understand this menu item also disappears when existing merchant visits the new setup wizard, however, say that I'm a merchant that has been using Square for many months/years, why would I click a "Square Onboarding" menu if they've already been connected and using Square?

My initial reaction when I saw this menu item was that clicking it was going to re-onboard me and so I can imagine merchants also being hesitant to click it themselves or being confused about what it's doing if they're an existing user of Square already.

I was thinking we should set `wc_square_onboarding_wizard_visited` to `1` by default for existing merchants unless there's a reason we want existing merchants to visit this new setup wizard?

This isn't a blocking issue and just my opinion so happy to move it to a separate issue to discuss it further or wait for feedback from merchants :)

> I checked and it does not appear to me. Can you please confirm if there is any PHP warning behind that white strip?

I cannot see any PHP warnings or notices behind the white strip.

From my latest testing, I can only reproduce it when the Square payment method is enabled, and Digital Wallets is enabled. So I'm guessing the problem is with trying to register my local domain with ApplePay on every page load of Square settings and it's erroring, maybe? I can't see anything in the logs that would suggest that though 🤔 

I just noticed I can reproduce the error on `trunk` so I think we move this off to a separate issue to investigate further or ignore it since it's most likely a local dev issue only :)

faisal-alvi committed:
mark the onboarding wizard as visited for existing users

faisal-alvi commented:
> My understanding of these two settings might be incorrect now though so I'm happy for you to confirm and decide on whether a description is needed.

We have internally decided to merge these settings together rather than keeping them separate as they seem to do more or less the same thing and we'll do that in a separate PR.

> I was thinking we should set wc_square_onboarding_wizard_visited to 1 by default for existing merchants unless there's a reason we want existing merchants to visit this new setup wizard?

Thanks for the suggestion. I was thinking of showing the new onboarding at least once even to the existing users as all users (new/existing) can visit the onboarding from the plugin list page as we have added the wizard link right beside the deactivate button. However, I have implemented your suggestion and set `wc_square_onboarding_wizard_visited` to 1 by default for existing merchants.

> I just noticed I can reproduce the error on trunk so I think we move this off to a separate issue

👍🏻 makes sense. Can you please create an issue ticket with the details?

vikrampm1 commented:
@mattallan can you review the updates from Faisal above and see if we can proceed moving this PR to the next step?

faisal-alvi committed:
load assets on onboarding only

faisal-alvi committed:
Merge branch 'trunk' into wizard_page

mattallan reviewed (approved):
> we'll do that in a separate PR.

Sounds good! Thanks @faisal-alvi 

> I have implemented your suggestion and set wc_square_onboarding_wizard_visited to 1 by default for existing merchants.

Thanks for making this change as well. I think this will avoid a lot of confusion to new merchants so I'm happy to see this one :) I have tested:

- [x] upgrading to this PR as an existing merchant and confirm the onboarding is not shown
- [x] new merchants still see the onboarding wizard

> Can you please create an issue ticket with the details?

Yea, I'll do some more investigating tomorrow and open an issue.

---

I've gone through the onboarding wizard again on a fresh site and confirmed it's looking good. 
Happy to approve this. Thanks @Sidsector9 and @faisal-alvi for all the work on this one

jeffpaul commented:
> > My understanding of these two settings might be incorrect now though so I'm happy for you to confirm and decide on whether a description is needed.

> We have internally decided to merge these settings together rather than keeping them separate as they seem to do more or less the same thing and we'll do that in a separate PR.

@faisal-alvi @vikrampm1 please ensure that issue is created and linked to the source conversation here so we can track that, thanks!

vikrampm1 commented:
@jeffpaul yes it's already created and linked/referenced with this PR: https://github.com/woocommerce/woocommerce-square/issues/130

faisal-alvi committed:
fix modal content size

faisal-alvi committed:
add `!` to changes saved text

faisal-alvi committed:
show changes saved text, not just the icon

faisal-alvi committed:
Enable Sandbox container "gray" is a bit dark compared to Figma

faisal-alvi committed:
Add validation when sandbox fields are empty

faisal-alvi committed:
update the save btn text to 'Connected to Sandbox'

faisal-alvi committed:
make business location selected when only one exists

faisal-alvi committed:
Add a "back" icon for payment gateway settings

faisal-alvi committed:
Change "Enable this gateway" to "Enable this payment method"

faisal-alvi committed:
Remove old settings fields instead of hiding them with CSS

faisal-alvi committed:
Fix button colors to match admin theme colors

faisal-alvi committed:
fix the primary button designs

Sidsector9 committed:
add message when sandbox not connected in onboarding

faisal-alvi committed:
fix the business location unset issue

faisal-alvi committed:
button busy state as per theme

faisal-alvi committed:
changed saved text on save

faisal-alvi committed:
render warning fix

faisal-alvi committed:
a bit visible border on required field

faisal-alvi committed:
show location when sandbox connected

faisal-alvi committed:
"above the fold" wrap + improved consistency

faisal-alvi committed:
how to get merchants back into the setup flow?

faisal-alvi committed:
connect to square updated section

faisal-alvi committed:
increased fonts

faisal-alvi committed:
Gift card checkbox UI

faisal-alvi committed:
phpcs fixes

faisal-alvi committed:
fix URL issue when multisite network enabled

faisal-alvi committed:
reload settings page on save to populate updated locations

faisal-alvi committed:
fix phpcs issues,
FILE: includes/Admin.php
-----------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 2 LINES
-----------------------------------------------------------------------------------------------
 240 | WARNING | Array double arrow not aligned correctly; expected 4 space(s) between "'nonce'" and double arrow, but found 1. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
 274 | WARNING | Array double arrow not aligned correctly; expected 4 space(s) between "'nonce'" and double arrow, but found 1. (WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned)
-----------------------------------------------------------------------------------------------

faisal-alvi committed:
keep showing message/submenu until connected page visited

faisal-alvi committed:
minor phpcs issues fix,
FILE: includes/AJAX.php
-----------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 1 WARNING AFFECTING 2 LINES
-----------------------------------------------------------------------------------------------
  82 | ERROR | Whitespace found at end of line (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 187 | WARNING | The method parameter $request is never used (Generic.CodeAnalysis.UnusedFunctionParameter.Found)
-----------------------------------------------------------------------------------------------

faisal-alvi committed:
consistent naming on payments page

faisal-alvi committed:
suggestions implemented

dkotter commented:
@Sidsector9 @faisal-alvi There's quite a few merge conflicts here. Can you help get those cleaned up so this can move on to the next step?

faisal-alvi committed:
Merge branch 'trunk' into wizard_page

faisal-alvi committed:
updates in eslint package to resolve dep conflicts

faisal-alvi committed:
bump back `@wordpress/scripts` to resolve ESlint issue,
https://github.com/woocommerce/woocommerce-square/actions/runs/9594437511/job/26456998876

faisal-alvi committed:
update package lock file

Sidsector9 committed:
Merge branch 'wizard_page' into wizard_page_extras

faisal-alvi committed:
bump back `eslint-plugin-prettier` to fix eslint workflow

faisal-alvi committed:
fix eslint across 45+ files

faisal-alvi committed:
a few more lint fixes

faisal-alvi committed:
Merge branch 'wizard_page' into wizard_page_extras

faisal-alvi committed:
eslint fixes

faisal-alvi committed:
more linting fixes

faisal-alvi committed:
a few changes after conflicts resolve

faisal-alvi committed:
home url fix for multisite network

qasumitbagthariya commented:
Regression / Smoke Test Report  ✅ 
-------

Tested with Archive File created via "php woorelease.phar build repo_URL" (Composer version 2.5.5, npm version 8.19.4, node version 16.20.0)

Status- Working expected with Plugin Archive/Zip file same as fix specific branch.

**Testing Environment** 

<Details>

- WordPress: 6.5.4
- Theme: Twenty Twenty-Four 1.1
- WooCommerce - 9.0.1
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Git Branch: smoke-testing

</Details>


Next Step- Ready to Merge 🚀
